src/test/clojure/clara/test_compiler.clj

```diff
@@ -55,4 +57,22 @@
                                          "-"
                                          (:id node)))
                         (-> node-fn str m/demunge (str/split #"/") last)))
-          (str "For node: " node " and node-fn: " node-fn)))))
\ No newline at end of file
+          (str "For node: " node " and node-fn: " node-fn)))))
+
+;; See https://github.com/cerner/clara-rules/issues/454 for more info
+(deftest test-query-node-requires-bindings-exist
+  (let [;; (defquery a-query
+        ;;   [:?b]
+        ;;   [?c <- ::a-fact-type])
+        query {:lhs [{:type ::a-fact-type
+                      :constraints []
+                      :args []
+                      :fact-binding :?c}]
+               :params #{:?b}
+               :name "a-query"}]
+    (try (com/mk-session [[query]])
+         (catch ExceptionInfo exc
```

Comment:
It looks like you could use [assert-ex-data](https://github.com/cerner/clara-rules/blob/0.20.0/src/main/clojure/clara/tools/testing_utils.cljc#L192) here to reduce brittleness in case this exception was ever caught and wrapped.

Reply:
+1 to the changes in https://github.com/cerner/clara-rules/pull/455/commits/b559d93351f34cd058a05435b3a32dafa926bcb9